README.md

```diff
@@ -38,90 +38,133 @@
 
-Detailed and narrated [examples](https://github.com/stellargraph/stellargraph/tree/master/demos/) of various machine learning workflows on network data, supported by StellarGraph, from data ingestion into graph structure to inference, are given in the `demos` directory of this repository.
+#### Graph machine learning model ####
 
-<!--
-StellarGraph supports different machine learning use-cases, including:
+# Use StellarGraph to create the graph machine learning model: graph, data generator, and GCN layers
+graph = sg.StellarGraph(nodes, edges)
 
-* Representation learning for nodes
-  - See the demos in folder `demos/embeddings` for examples of unsupervised node representation learning using the
-  random walk-based methods Node2Vec [1], and Metapath2Vec [2].
+generator = sg.mapper.FullBatchNodeGenerator(graph, method="gcn")
+train_gen = generator.flow(train_targets.index, train_targets)
 
-* Node classification and regression
-  - See the demo in folder `demos/node-classification-graphsage` for an example of how to predict attributes of nodes
-  using the GraphSAGE [3] algorithm given node features and training labels.
-  - See the demo in folder `demos/node-classification-node2vec` for an example of how to predict attributes of nodes
-  using the Node2Vec [1] algorithm for nodes without features, unsupervised node representation learning, and
-  supervised classifier training for the downstream task.
-  - See the demo in folder `demos/node-classification-hinsage` for examples of how to predict attributes of nodes
-  using the HinSAGE algorithm for given node features and training labels.
+# two layers of GCN, each with hidden dimension 16
+gcn = sg.layer.GCN(
+    layer_sizes=[16, 16], activations=["relu", "relu"], generator=generator, dropout=0.5
+)
+x_inp, x_out = gcn.build() # input and output tensors
 
-* Link prediction
-  - See the demo in folder `demos/link-prediction-random-walks` for an example of how to predict the existence of links between nodes
-  without node features, using the Node2Vec [1] and Metapath2Vec [2] algorithms.
-  - See the demo in folder `demos/link-prediction-graphsage` for an example of how to predict the existence of links between
-  nodes with node features using the GraphSAGE [3] algorithm.
+# use TensorFlow Keras to add a layer to compute the (one-hot) predictions
+predictions = tf.keras.layers.Dense(units=len(ground_truth_targets.columns), activation="softmax")(x_out)
 
-* Recommender systems
-  - See the demo in folder `demos/link-prediction-hinsage` for an example of how to predict
-  movie ratings between users and movies using a Heterogeneous generalisation of GraphSAGE model, which we call HinSAGE.
+#### Training and prediction ####
 
--->
+# Use TensorFlow Keras to create a model and work with it  (e.g. training and evaluation)
+model = tf.keras.Model(inputs=x_inp, outputs=predictions)
+model.compile("adam", loss="categorical_crossentropy", metrics=["accuracy"])
 
+model.fit(train_gen, epochs=5)
```

Comment:
Thoughts on calling `generator.flow` inline here, similar to `evaluate`?
```suggestion
model.fit(generator.flow(train_targets.index, train_targets), epochs=5)
```
Seems like would be nice if we supported `generator.flow(train_targets)` using pandas dataframe index by default actually